apps/wpcom-block-editor/src/wpcom/features/tracking.js

```diff
@@ -110,16 +110,42 @@ const getBlocksTracker = ( eventName ) => ( blockIds ) => {
 	} );
 };
 
+/**
+ * Determines whether a block pattern has been inserted and if so, records
+ * a track event for it. The recorded event will also reflect whether the
+ * inserted pattern replaced blocks.
+ *
+ * @param {Array} args Data supplied to block insertion or replacement tracking functions.
+ * @returns {string} Pattern name being inserted if available.
+ */
+const trackPatternInsertion = ( args ) => {
+	const meta = find( args, ( arg ) => arg?.patternName );
+	const patternName = meta?.patternName;
+
+	if ( patternName ) {
+		tracksRecordEvent( 'wpcom_pattern_inserted', {
+			pattern_name: patternName,
+			blocks_replaced: args.blocks_replaced,
```

Comment:
Is it guaranteed that `args.blocks_replaced,` is always defined? If not then the following might be safer...

```js
args?.blocks_replaced,
```